src/core/CompUnit/RepositoryRegistry.pm6

```diff
@@ -96,7 +96,17 @@ class CompUnit::RepositoryRegistry {
                      ?? nqp::atkey($ENV,'HOMEPATH') !! '')
                  ) -> $home-path {
                 $home = "$home-path/.perl6";
-                my str $path = "inst#$home";
+                nqp::mkdir($home, 0o700) unless nqp::stat($home, nqp::const::STAT_EXISTS);
```

Comment:
Are we sure we want this side effect?

Reply:
Do you have any better suggestion to figure out if the home directory can be written to?